pkg/scheduler/framework/plugins/noderesources/least_allocated.go

```diff
@@ -63,25 +64,45 @@ func (la *LeastAllocated) ScoreExtensions() framework.ScoreExtensions {
 }
 
 // NewLeastAllocated initializes a new plugin and returns it.
-func NewLeastAllocated(_ runtime.Object, h framework.FrameworkHandle) (framework.Plugin, error) {
+func NewLeastAllocated(laArgs runtime.Object, h framework.FrameworkHandle) (framework.Plugin, error) {
+	args, err := getNodeResourcesLeastAllocatedArgs(laArgs)
+	if err != nil {
+		return nil, err
+	}
+
+	resToWeightMap := make(resourceToWeightMap)
+	for _, resource := range args.Resources {
+		resToWeightMap[v1.ResourceName(resource.Name)] = resource.Weight
+	}
+
 	return &LeastAllocated{
 		handle: h,
 		resourceAllocationScorer: resourceAllocationScorer{
-			LeastAllocatedName,
-			leastResourceScorer,
-			defaultRequestedRatioResources,
+			Name:                LeastAllocatedName,
+			scorer:              leastResourceScorer(resToWeightMap),
+			resourceToWeightMap: resToWeightMap,
 		},
 	}, nil
 }
 
-func leastResourceScorer(requested, allocable resourceToValueMap, includeVolumes bool, requestedVolumes int, allocatableVolumes int) int64 {
-	var nodeScore, weightSum int64
-	for resource, weight := range defaultRequestedRatioResources {
-		resourceScore := leastRequestedScore(requested[resource], allocable[resource])
-		nodeScore += resourceScore * weight
-		weightSum += weight
+func getNodeResourcesLeastAllocatedArgs(obj runtime.Object) (config.NodeResourcesLeastAllocatedArgs, error) {
+	ptr, ok := obj.(*config.NodeResourcesLeastAllocatedArgs)
+	if !ok {
+		return config.NodeResourcesLeastAllocatedArgs{}, fmt.Errorf("want args to be of type NodeResourcesLeastAllocatedArgs, got %T", obj)
```

Comment:
nit: changing return type to pointer would make this line shorter.

Also, we put this code inside a function when we used to set the defaults in it. Now it's just simpler to put this code in `NewLeastAllocated`

Reply:
done